doc/mgmt/show_running_cfg.md

```diff
@@ -0,0 +1,495 @@
+
+## **SONiC Management Framework CLI 'Show Running Configuration'**
+
+
+
+#### High Level Design Document
+
+**Rev 0.1**
+
+#### Table of Contents
+
+-   [[List of Tables]](#list-of-tables)
+
+-   [[Revision]](#revision)
+
+-   [[About This Manual]](#about-this-manual)
+
+-   [[Scope]](#scope)
+
+-   [[Definition/Abbreviation]](#definitionabbreviation)
+
+#### List of Tables
+
+[[Table 1: Abbreviations]](#table-1-abbreviations)
+
+#### About this Manual
+
+This document states the functional requirements and high level design of the new feature to support "show running-configuration" command.
+
+#### Scope
+
+The scope of this document is limited to the functionality described in the
+requirements section.
+
+#### Definition/Abbreviation
+
+NA
+
+
+
+#### 1 Feature Overview
+
+This feature provides an infrastructure for apps to add
+their current system configuration to the output of the "show
+running-configuration" command. This command is available at the exec
+level on the SONiC management framework CLI.
+
+#### 1.1 Functional requirements
+
+The 'show running-configuration' command will display the  configuration done
+through the CLI interface and which is stored on the system.
+
+For this command to show the entire running configuration of the system,
+all apps should support configuration through CLI interface.
+
+Any configuration done using GNMI or REST or other management
+interface which is stored and has a corresponding CLI
+command will be displayed as part of this new command.
+
+The command output can be copied and pasted to be used as
+configuration on another SONiC CLI session of the same or another
+system.
+
+The order of commands displayed in the "show running-configuration" will be
+in accordance to the order of dependency between CLI views.
+
+If configuration default values are stored, they will be shown as
```

Comment:
Why do we want to show all default configs under show running-config?
It will not be readable and i believe it is not inline with industry standards.